dorado/particle_track.py

```diff
@@ -403,14 +373,160 @@ def __init__(self, params):
+
+            previous_walk_data : `dict`, optional
+                Dictionary of all prior x locations, y locations, and travel
+                times. This input parameter should only be used if 'new' or
+                'indpendent' walk data exists (e.g. data loaded from a .json
+                file). Otherwise the walk data stored in `self.walk_data` is
+                likely to contain the previous information. Order of indices is
+                previous_walk_data[field][particle][iter], where e.g.
+                ['travel_times'][5][10] is the travel time of the 5th particle
+                at the 10th iteration.
+
+        """
+        # if the values in self are invalid the input checked will catch them
+        # do input type checking
+        Np_tracer, seed_xloc, seed_yloc = gen_input_check(Np_tracer,
+                                                          seed_xloc,
+                                                          seed_yloc,
+                                                          method)
+
+        init_walk_data = dict()  # create init_walk_data dictionary
+
+        # initialize new travel times list
+        new_start_times = [0.]*Np_tracer
+
+        if method == 'random':
+            # create random start indices based on x, y locations and np_tracer
+            new_start_xindices = [lw.random_pick_seed(seed_xloc) for x
+                                  in list(range(Np_tracer))]
+            new_start_yindices = [lw.random_pick_seed(seed_yloc) for x
+                                  in list(range(Np_tracer))]
+
+        elif method == 'exact':
+            # step through seed lists to create exact list of start locations
+            # if number of particles exceeds length of seed lists, loop back
+            # to beginning of the seed lists and continue from there
+            Np_to_seed = Np_tracer
+            new_start_xindices = []
+            new_start_yindices = []
+            while Np_to_seed > 0:
+                for i in range(0, len(seed_xloc)):
+                    # condition if we run out of particles within this loop
+                    if Np_to_seed <= 0:
+                        break
+                    # otherwise keep on seeding
+                    new_start_xindices = new_start_xindices + [seed_xloc[i]]
+                    new_start_yindices = new_start_yindices + [seed_yloc[i]]
+                    Np_to_seed -= 1  # reduce number of particles left to seed
+
+        # Now initialize vectors that will create the structured list
+        new_xinds = [[new_start_xindices[i]] for i in
+                     list(range(Np_tracer))]
+        new_yinds = [[new_start_yindices[i]] for i in
+                     list(range(Np_tracer))]
+        new_times = [[new_start_times[i]] for i in list(range(Np_tracer))]
+
+        # establish the start values
+        start_xindices = new_xinds
+        start_yindices = new_yinds
+        start_times = new_times
```

Comment:
Looks like the newest particles being generated get assigned the first indices in `walk_data`, regardless of whether other particles exist. It seems like we may want to reorder the steps here so that either `previous_walk_data` or `self.walk_data` retain first priority in the `walk_data` indexing. Not sure which order is ideal, but my guess is we would want `previous_walk_data` particles to be first (likely a previous, finished particle run), `self.walk_data` particles second (new particles, but slightly older than the ones actively being created), then `new_` particles last (as they are the freshest on the scene).

Reply:
Sure, they are just lists so we can change the order in which they are appended.